examples/Viewer.tsx

```diff
@@ -208,17 +222,83 @@ class Viewer extends React.Component<{}, ViewerState> {
             .then((blob: Blob) => {
                 return loadSimulariumFile(blob);
             })
-            .then((simulariumFile) => {
+            .then((trajectoryFile) => {
                 const fileName = url;
-                simulariumController
-                    .changeFile({ simulariumFile }, fileName)
-                    .catch((error) => {
-                        console.log("Error loading file", error);
-                        window.alert(`Error loading file: ${error.message}`);
-                    });
+                this.loadFile(trajectoryFile, fileName).catch((error) => {
+                    console.log("Error loading file", error);
+                    window.alert(`Error loading file: ${error.message}`);
+                });
             });
     }
 
+    public async loadUiTemplates() {
+        const baseTypes = await fetch(
+            `${UI_TEMPLATE_DOWNLOAD_URL_ROOT}/${UI_BASE_TYPES}`
+        ).then((data) => data.json());
+        const customTypes = await fetch(
+            `${UI_TEMPLATE_URL_ROOT}/${UI_CUSTOM_TYPES}`
+        )
+            .then((data) => data.json())
+            .then((fileRefs) =>
+                Promise.all(
+                    map(fileRefs, (ref) =>
+                        fetch(ref.download_url).then((data) => data.json())
+                    )
+                )
+            );
+        const typeMap = reduce(
+            customTypes,
+            (acc, cur) => {
+                const key = Object.keys(cur)[0];
+                acc[key] = cur[key];
+                return acc;
+            },
+            {}
+        );
+        baseTypes["base_types"].forEach((type) => {
+            typeMap[type.id] = { ...type, isBaseType: true };
+        });
+        return typeMap;
```

Comment:
would love to see a type description for typeMap to know what it's supposed to look like